src/Compilers/Core/Portable/CommandLine/ErrorLogger.cs

```diff
@@ -197,39 +199,38 @@ private Value GetSpanInfoValue(FileLinePositionSpan lineSpan)
             return Value.Create(builder.ToImmutableAndFree(), this);
         }
 
-        private Value GetPropertiesValue(Issue issue)
+        private Value GetPropertiesValue(Result result)
         {
             var builder = ArrayBuilder<KeyValuePair<string, Value>>.GetInstance();
 
-            builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.Severity, issue.Severity.ToString()));
-            if (issue.Severity == DiagnosticSeverity.Warning)
+            builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.Severity, result.Severity.ToString()));
+            if (result.Severity == DiagnosticSeverity.Warning)
             {
-                builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.WarningLevel, issue.WarningLevel.ToString()));
+                builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.WarningLevel, result.WarningLevel.ToString()));
             }
 
-            builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.DefaultSeverity, issue.DefaultSeverity.ToString()));
+            builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.DefaultSeverity, result.DefaultSeverity.ToString()));
 
-            if (!string.IsNullOrEmpty(issue.Title))
+            if (!string.IsNullOrEmpty(result.Title))
             {
-                builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.Title, issue.Title));
+                builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.Title, result.Title));
             }
 
-            builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.Category, issue.Category));
+            builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.Category, result.Category));
 
-            if (!string.IsNullOrEmpty(issue.HelpLink))
+            if (!string.IsNullOrEmpty(result.HelpLink))
             {
-                builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.HelpLink, issue.HelpLink));
+                builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.HelpLink, result.HelpLink));
             }
 
-            builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.IsEnabledByDefault, issue.IsEnabledByDefault.ToString()));
-            builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.IsSuppressedInSource, issue.IsSuppressedInSource.ToString()));
+            builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.IsEnabledByDefault, result.IsEnabledByDefault.ToString()));
 
-            if (issue.CustomTags.Count > 0)
+            if (result.CustomTags.Count > 0)
             {
-                builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.CustomTags, issue.CustomTags.WhereNotNull().Join(";")));
+                builder.Add(CreateSimpleKeyValuePair(WellKnownStrings.CustomTags, result.CustomTags.WhereNotNull().Join(";")));
```

Comment:
I'm not sure I agree. We originally defined two kinds of storage for custom data on a result. A dictionary of string/string key/value pairs, and a tags property, comprised of an array of distinct string values. We made a design choice to drop the tags member, because the properties member could be repurposed for this. If we stand by that call, we should be explicit there. It is not helpful for format producers to create their own serialization mechanism within dictionary members and we should advise against it. Consider "MyDictionary" = "key1=foo;key2=bar".

I think we should consider whether we reintroduce a tags member, so that producers fall into the pit of success of using it. If we do not, we should advise producers explicitly on how to handle custom tags.

Part of the value of SARIF, for the conversion scenario, is that it attempts to minimize special knowledge of a particular producer as far as possible. If a producer bases a special delimiter/other custom serialization scheme, a SARIF consumer needs to know about that to make sense of the information. 'That is Bad'. On the other hand, if a producer breaks apart its tags in the spec'ed way, this can be reasonably handled. These comments only apply because a property bag or a custom tag set may, arguably, drive some generalizer user experiences (such as dumping all reported tags into an error list column).